dev-lang/rapydscript-ng/rapydscript-ng-0.7.22.ebuild

```diff
@@ -0,0 +1,72 @@
+# Copyright 2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+if [[ ${PV} == 9999 ]] ; then
+	EGIT_REPO_URI="https://github.com/kovidgoyal/rapydscript-ng"
+	# this is nodejs software, so possibly unfixable. I am sorry.
+	# Instead, we will have to generate this with the live ebuild and upload assets
+	# using EGIT_OVERRIDE_COMMIT for the tagged version we need.
+	RESTRICT="network-sandbox"
+	BDEPEND="app-arch/libarchive"
+
+	inherit git-r3
+else
+	SRC_URI="
+		https://github.com/kovidgoyal/rapydscript-ng/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz
+		https://help-me/rapydscript-ng-node_modules-${PV}.tar.xz
+	"
+	KEYWORDS="~amd64"
+fi
+
+inherit edo
+
+DESCRIPTION="Pythonic JavaScript that doesn't suck"
+HOMEPAGE="https://github.com/kovidgoyal/rapydscript-ng"
+
+LICENSE="BSD"
+SLOT="0"
+
+DEPEND="net-libs/nodejs"
+RDEPEND="${DEPEND}"
+
+maint_pkg_create() {
+	cd "${S}" || die
+
+	edo npm install --omit=optional
+	local ver=$(git describe)
+	ver=${ver#v}
+	local tar="${T}/rapydscript-ng-node_modules-${ver}.tar.xz"
+
+	bsdtar -s "#\(.*\)#${PN}-node_modules-${ver}/~#S" -caf "${tar}" package-lock.json node_modules/ || die
+	einfo "Packaged tar now available:"
+	einfo "$(du -b "${tar}")"
+
+}
+
+src_unpack() {
+	if [[ ${PV} == 9999 ]] ; then
+		git-r3_src_unpack
+		maint_pkg_create
+	else
+		unpack ${A}
+		cp -r "${WORKDIR}"/${PN}-node_modules-${PV}/{node_modules,package-lock.json} "${S}" || die
+	fi
+}
+
+src_compile() {
+	edo bin/rapydscript self --complete
+	rm -r release/ || die
+	mv dev/ release/ || die
+}
+
+src_test() {
+	edo bin/rapydscript test
+}
+
+src_install() {
+	mkdir -p "${ED}/usr/$(get_libdir)"/node_modules/rapydscript-ng/ || die
+	cp -r *  "${ED}/usr/$(get_libdir)"/node_modules/rapydscript-ng/ || die
+	dosym ../$(get_libdir)/node_modules/rapydscript-ng/bin/rapydscript /usr/bin/rapydscript
```

Comment:
EAPI 8 lets us do `dosym -r` and avoid the `..`.

Reply:
Add that to "things which need to be integrated with https://devmanual.gentoo.org/function-reference/install-functions/" then. :)

Reply:
I'd mentioned it at https://bugs.gentoo.org/914590#c2 but it got lost, I think. I'll just file a new bug for it.